staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec_test.go

```diff
@@ -144,6 +145,48 @@ func TestCacheKey(t *testing.T) {
 	}
 }
 
+func TestCacheOldestExpiration(t *testing.T) {
+	c := newCache()
+	if !c.oldestExpiration().IsZero() {
+		t.Error("empty cache should return a zero time")
+	}
+	newCert, _ := tls.X509KeyPair(genClientCert(t))
+	newAuth := &Authenticator{cachedCreds: &credentials{cert: &newCert}}
+	// validCert has a NotAfter of Jan 1 2016
+	oldAuth := &Authenticator{cachedCreds: &credentials{cert: validCert}}
+	additiveTests := []struct {
+		a *Authenticator
+		e time.Time
+	}{
+		{
+			a: &Authenticator{cachedCreds: &credentials{token: "friend"}},
+			e: time.Time{}, // Zero
+		},
+		{
+			a: &Authenticator{cachedCreds: &credentials{token: "password"}},
+			e: time.Time{}, // Zero
+		},
+		{
+			a: newAuth,
+			e: newAuth.leafCertificate().NotAfter,
+		},
+		{
+			a: oldAuth,
+			e: oldAuth.leafCertificate().NotAfter,
+		},
+	}
+	for i, tc := range additiveTests {
+		// Must run in series as the tests build off each other.
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
```

Comment:
add actual descriptions to test cases instead of using index

Reply:
done